tests/unit/s2n_client_psk_extension_test.c

```diff
@@ -1543,5 +1545,128 @@ int main(int argc, char **argv)
+     **/
+    {
+        DEFER_CLEANUP(struct s2n_cert_chain_and_key *chain_and_key,
+                      s2n_cert_chain_and_key_ptr_free);
+        EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
+                                                       S2N_DEFAULT_ECDSA_TEST_CERT_CHAIN,
+                                                       S2N_DEFAULT_ECDSA_TEST_PRIVATE_KEY));
+
+        DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(),
+                      s2n_config_ptr_free);
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, chain_and_key));
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "default_tls13"));
+        EXPECT_SUCCESS(s2n_config_disable_x509_verification(config));
+        EXPECT_SUCCESS(s2n_setup_ticket_key(config));
+
+        DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER),
+                      s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(server_conn);
+        EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
+        struct s2n_stuffer *server_io = &server_conn->handshake.io;
+
+        DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT),
+                      s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(client_conn);
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config));
+        struct s2n_stuffer *client_io = &client_conn->handshake.io;
+
+        struct s2n_test_io_pair io_pair = { 0 };
+        EXPECT_SUCCESS(s2n_io_pair_init_non_blocking(&io_pair));
+        EXPECT_SUCCESS(s2n_connections_set_io_pair(client_conn, server_conn, &io_pair));
+
+        /* Force the HRR path */
+        client_conn->security_policy_override = &security_policy_test_tls13_retry;
+
+        client_conn->actual_protocol_version = S2N_TLS13;
+        client_conn->secure.cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
+
+        /* Set a resumption psk */
+        DEFER_CLEANUP(struct s2n_stuffer psk_identity = { 0 }, s2n_stuffer_free);
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&psk_identity, 0));
+        EXPECT_OK(s2n_setup_encrypted_ticket(client_conn, &psk_identity));
+        struct s2n_offered_psk_list identity_list = { .conn = client_conn };
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&identity_list.wire_data, 0));
+        EXPECT_OK(s2n_write_test_identity(&identity_list.wire_data, &psk_identity.blob));
+        EXPECT_OK(s2n_select_resumption_psk(server_conn, &identity_list));
+
+        /* Add arbitrary amount to ticket age so a non-zero value implies it's been processed */
+        struct s2n_psk *psk = client_conn->psk_params.chosen_psk;
+        psk->ticket_age_add = 10;
+
+        /* ClientHello 1 */
+        EXPECT_SUCCESS(s2n_client_hello_send(client_conn));
+
+        /* Read the obfuscated ticket age from ClientHello 1 */
+        EXPECT_SUCCESS(s2n_stuffer_skip_read_until(client_io, (char*) psk->identity.data));
```

Comment:
Because `s2n_stuffer_skip_read_until` includes the target in the amount it increments the read position, and this amount is not the full identity size due to `s2n_stuffer_skip_read_until` using strlen to check for the size. The ticket key name is null terminated, so seeking until `identity.data` includes just the key name. So, I skipped until the key name and skipped backwards by the key name length so I could skip forward by the entire key length amount.

I removed the seek backwards and just seeked forwards by the key length - the key name in 391bef95511f6bb8d8e92f9d1eff39fbc966bc2d. Let me know if this is more clear!

Reply:
You should probably move (psk->identity.size - strlen((char*) psk->identity.data)) to a named variable with a comment explaining why it exists :) Then you can just refer to that variable.

Reply:
Fixed! 2c0f94d8712641186acdd45b13e277dfc03bce88